tests/Writer/CategoriesTest.php

```diff
@@ -19,22 +19,46 @@ public function test_categories_xml(): void
         $xml->startDocument();
         $xml->startCatalog('TestCatalog');
 
-        foreach (['Socks', 'Death Stars', 'Donuts'] as $index => $example) {
-            $element = new Category('CAT' . $index);
-            $element->setDisplayName($example);
+        $categories = [
+            'Socks' => [
+                'index' => 0,
+                'dates' => [
+                    new DateTimeImmutable('2018-01-01 01:01:01'),
```

Comment:
Be nice to give these arrays keys (`from` and `to`).

Reply:
Sorted